sdk/cloudmachine/Azure.CloudMachine/src/CloudMachineClient.cs

```diff
@@ -27,6 +29,11 @@ public partial class CloudMachineClient : ClientWorkspace
     [EditorBrowsable(EditorBrowsableState.Never)]
     public ConnectionCollection Connections { get; } = [];
 
+    private readonly ReaderWriterLockSlim _connectionsLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+    private readonly ReaderWriterLockSlim _connectionCacheLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+    private readonly Dictionary<ConnectionType, ConnectionResponse> _connectionCache = new Dictionary<ConnectionType, ConnectionResponse>();
+    private ConnectionsClient _foundryConnectionsClient;
```

Comment:
CloudMachineCLient should not have a hard dependency on AI Project Clients. The AIFoundryFeature would have a dependency on it and when CMClient is created from the infrastrucutre object, all connections are read and transferred to the CMCLient

Reply:
Fixed this by moving all the AI Projects connection logic into AIFoundryFeature